### DISCUSSION TITLE
Multiple clients using the same instance of a plugin

### DISCUSSION THREAD

**Discussion by jtayl711** [apohl79/audiogridder#1151] [Q&A]
Is it possible to have multiple clients run different instances of the same plugin from the server, and have the audio from each instance be routed on separate channels?  I have a setup where I have a main computer that can host all my virtual instruments, but I have several musicians, each with their own laptop, on a LAN.  I'd like them to each be able to control separate instances of, let's say, the Surge Synthesizer, but have all the audio go back to a single computer.  So all the GUIs would run on separate computers, but all the DSP would be happening on a single computer for each instance.  Not sure if I've explained that well.

**Comment by oldtos** [comment]
Hmmm.. uncharted for me... but.... running more then 1 server on the server computer is possible... .. so .. ( not sure) if each person connected to a different server instance ..  heck.. maybe :)

**Comment by apohl79** [comment]
As I guess you need each artist to use the plugin UI, you would need a separate UI session (like a separate VNC login to the computer with individual users on the operating system) per artist. On each session/vnc login you would run an AG server with a different ID. And each artist would now connect to one of the AG servers.

**Reply by jtayl711** [reply]
Great I'll give this a shot, thank you.  Just to confirm I'm understanding correctly...

- On the server PC set up multiple windows users (1 for each 'musician')
- On the server PC, spin up a server for each musician
- On the client side, each musician would use a unique vnc session to log in to their instance of the server
- On the client side, musician A, logged into their vnc session, could start an instance of plugin 'XYZ'
- On the client side, musician B, logged into their vnc session, could also start an instance of plugin 'XYZ'
- Musicians A & B, while running separate instances of the same plugin, have unique control and view of each of their instances
Is that correct?

**Reply by apohl79** [reply]
> Great I'll give this a shot, thank you.  Just to confirm I'm understanding correctly...
> 
> 
> 
> - On the server PC set up multiple windows users (1 for each 'musician')
> 
> - On the server PC, spin up a server for each musician
> 
> - On the client side, each musician would use a unique vnc session to log in to their instance of the server

yes that would work. you could autostart an ag server upon login.

> 
> - On the client side, musician A, logged into their vnc session, could start an instance of plugin 'XYZ'

yes. each musician had to connect to 'their' ag server using their server id.

> 
> - On the client side, musician B, logged into their vnc session, could also start an instance of plugin 'XYZ'

yes 

> 
> - Musicians A & B, while running separate instances of the same plugin, have unique control and view of each of their instances
> 
> Is that correct?

exactly

**Reply by jtayl711** [reply]
I'll give it a shot thank you!

**Reply by wpostma** [reply]
Since the point of AudioGridder is to INCREASE your total DSP power, not sure what the point of dividing one audio server machine's power by 2 or 4 is. What's the benefit?
